presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -948,32 +953,94 @@ public void alterDatabase(MetastoreContext metastoreContext, String databaseName
+                                getMetastoreClientThenCall(metastoreContext, client -> {
+                                    client.createTable(table);
+                                    return null;
+                                })));
+            }
+            catch (AlreadyExistsException e) {
+                throw new TableAlreadyExistsException(new SchemaTableName(table.getDbName(), table.getTableName()));
+            }
+            catch (NoSuchObjectException e) {
+                throw new SchemaNotFoundException(table.getDbName());
+            }
+            catch (TException e) {
+                throw new PrestoException(HIVE_METASTORE_ERROR, e);
+            }
+            catch (Exception e) {
+                throw propagate(e);
+            }
         }
-        catch (Exception e) {
-            throw propagate(e);
+        else {
+            List<SQLPrimaryKey> primaryKeys = new ArrayList<>();
+            List<SQLUniqueConstraint> uniqueConstraints = new ArrayList<>();
+            for (TableConstraint<String> constraint : constraints) {
+                int keySeq = 1;
+                if (constraint instanceof PrimaryKeyConstraint) {
+                    for (String column : constraint.getColumns()) {
+                        primaryKeys.add(
+                                new SQLPrimaryKey(table.getDbName(),
+                                        table.getTableName(),
+                                        column,
+                                        keySeq++,
+                                        constraint.getName().orElse(null),
+                                        constraint.isEnable(),
+                                        constraint.isValidate(),
+                                        constraint.isRely()));
+                    }
+                }
+                else if (constraint instanceof UniqueConstraint) {
+                    for (String column : constraint.getColumns()) {
+                        uniqueConstraints.add(
+                                new SQLUniqueConstraint(
+                                        table.getCatName(),
+                                        table.getDbName(),
+                                        table.getTableName(),
+                                        column,
+                                        keySeq++,
+                                        constraint.getName().orElse(null),
+                                        constraint.isEnable(),
+                                        constraint.isValidate(),
+                                        constraint.isRely()));
+                    }
+                }
+                else {
+                    throw new PrestoException(NOT_SUPPORTED, format("Constraint %s of unknown type is not supported", constraint.getName().orElse("")));
+                }
+            }
+            try {
+                retry()
```

Comment:
this try/catch is almost the same as the one above except for what is passed to run(): can we refactor this into a function?